stdlib/LinearAlgebra/src/adjtrans.jl

```diff
@@ -267,10 +267,24 @@ Broadcast.broadcast_preserving_zero_d(f, tvs::Union{Number,TransposeAbsVec}...)
 
 ## multiplication *
 
+function dot_nonrecursive(u::AbstractVector, v::AbstractVector)
```

Comment:
I think you need to `permutedims` on the call site or just accept any `AbstractVecOrMat` here?

Reply:
Why?

Reply:
These `u`s and `v`s are not vectors

https://github.com/JuliaLang/julia/blob/8639f6ba5f0c065b662c44e103ee27036fab1ca6/stdlib/LinearAlgebra/src/adjtrans.jl#L298-L301

Reply:
I was thinking whether we should make this function "more internal" by prepending a `_` and leave the arguments unrestricted.

Reply:
Sounds good to me.

Reply:
Yep, makes sense.